tokio/src/runtime/context.rs

```diff
@@ -0,0 +1,305 @@
+//! Global runtime context
+#![allow(dead_code)]
+use crate::park::ParkThread;
+cfg_io_driver! {
+    use crate::io::driver::Handle as ReactorHandle;
+}
+cfg_rt_core! {
+    use crate::runtime::basic_scheduler::SchedulerPriv;
+}
+cfg_blocking_impl! {
+    use crate::runtime::blocking::Spawner;
+}
+cfg_rt_core! {
+    use crate::runtime::global::State as ExecutorState;
+}
+
+cfg_time! {
+    use crate::time::driver::Handle as TimerHandle;
+    use crate::time::Clock;
+}
+
+use std::cell::RefCell;
+
+thread_local! {
+    static CONTEXT: RefCell<ThreadContext> = RefCell::new(ThreadContext::new())
+}
+
+#[derive(Clone)]
+pub(crate) struct ThreadContext {
+    entered: bool,
```

Comment:
I think I'm wrong about this. It should be possible to set handle context w/o "entering".

We may want to rename `entered` here as it is kind of confusing. The ideal is to prevent "blocking" from an executor. Maybe this should be renamed to `executing`?